psa-crypto/src/types/key.rs

```diff
@@ -0,0 +1,768 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+//! # PSA Key types
+
+#![allow(deprecated)]
+
+use crate::types::algorithm::{Algorithm, Cipher};
+use crate::types::status::{status_to_result, Result, Status};
+use log::error;
+use serde::{Deserialize, Serialize};
+
+/// Native definition of the attributes needed to fully describe
+/// a cryptographic key.
+#[derive(Copy, Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct Attributes {
+    /// Lifetime of the key
+    pub lifetime: Lifetime,
+    /// Intrinsic category and type of the key
+    pub key_type: Type,
+    /// Size of the key in bits
+    pub bits: usize,
+    /// Policy restricting the permitted usage of the key
+    pub policy: Policy,
+}
+
+impl Attributes {
```

Comment:
Not part of this PR, but we should have a `validate(&self)` method here to check that all the fields are in a valid state (e.g. the `Type`, `Algorithm` and `bits` are in a consistent state.

Reply:
Would be good indeed!
We can also think about making some of those fields (all) private and creating specific constructors or builders that only build consistent attributes.

Reply:
That sounds good, we could also have static pre-defined builders for common algorithms, for example.